tests/python/pants_test/backend/python/interpreter_selection_utils.py

```diff
@@ -107,3 +110,48 @@ def skip_unless_python27_and_python3_present(func):
 def skip_unless_python27_and_python36_present(func):
   """A test skip decorator that only runs a test method if python2.7 and python3.6 are present."""
   return skip_unless_all_pythons_present(PY_27, PY_36)(func)
+
+
+def skip_if_interpreter_is_any_python_version(*versions):
```

Comment:
I might change this one to just accept interpreter constraints directly and avoid any custom parsing. Since we're just converting them to interpreter constraints anyway, it seems like a good idea to use the same format people are already familiar with for interpreter constraints, and it seems to reduce the chance that something unexpected will happen and we accidentally skip more python versions than the user intended. We can do the `'=={}.*'.format(PY_2)` in any calls to this method instead of trying to figure it out here.

Reply:
I disagree with this because the API is meant to check if that major version is being used, and the API should not be anything more flexible or complex than that.

Rather than wanting us to allow arbitrary interpreter constraints like `@skip_if_interpreter_is_any_python_version(">=3.4,<3.7")`, I think it's better to be more explicit with `@skip_if_interpreter_is_any_python_version("3.4", "3.5", "3.6")`.

We should start with the simplest API possible, and only make it more flexible if there's a compelling reason we need it. Right now, we're trying to only solve the problem of `skip if python37`, not a problem like `skip if python34 up to python37`.

Further, it would be confusing to change what we expect the parameter for these functions to be, because it would diverge from the file's `skip_unless_pythonx_present` functions expectation.

Reply:
The last paragraph seems to be resolved by `'=={}.*'.format(PY_2)`. I was under the impression that typical restrictions that cause us to want to use interpreters or not aren't "this single interpreter version has a weird bug" but rather "these set of interpreters don't have the `async` keyword". I think I disagree with your statement that `@skip_if_interpreter_is_any_python_version(">=3.4,<3.7")` is less explicit, or at least, I consider it to be much more explicitly stating the intent of the interpreter restriction than repeating single python versions and forcing the reader to somehow understand what is being restricted here. I also consider it to be a much "simpler" API than having repeated decorators for different python version restrictions, all of which don't do anything you couldn't type yourself in fewer characters -- using repeated decorators for slight variations seems to be an overengineering of sorts here.